amethyst_renderer/src/pipe/pass.rs

```diff
@@ -156,6 +156,17 @@ impl<P> CompiledPass<P> {
         self.inner
             .apply(Supplier::new(encoders, &mut self.effects[..]), data)
     }
+
+    pub fn new_target(&mut self, target: &Target) {
+        for mut effect in &mut self.effects {
+            effect.data.out_colors.clear();
+            effect
```

Comment:
Couln't you just clone the other color_bufs directly into the effect one, instead of iterating through? (Not sure it would clone the inner elements, though)

Reply:
So yes, but it does come at the cost of an additional allocation, which I don't think is worth it.